actors/runtime/src/runtime/fvm.rs

```diff
@@ -263,28 +261,40 @@ where
         value: TokenAmount,
     ) -> Result<RawBytes, ActorError> {
         if self.in_transaction {
-            return Err(actor_error!(SysErrIllegalActor; "runtime.send() is not allowed"));
+            return Err(actor_error!(unspecified; "runtime.send() is not allowed"));
         }
-        // TODO: distinguish between "syscall" errors and actor exit codes.
         match fvm::send::send(&to, method, params, value) {
             Ok(ret) => {
                 if ret.exit_code.is_success() {
                     Ok(ret.return_data)
                 } else {
-                    Err(ActorError::from(ret.exit_code))
+                    // The returned code can't be simply propagated as it may be a system exit code.
+                    Err(ActorError::internal_send_aborted(format!(
+                        "send to {} method {} aborted with code {}",
+                        to, method, ret.exit_code
+                    )))
                 }
             }
             Err(err) => Err(match err {
+                // Some of these errors are from operations in the Runtime or SDK layer
+                // before or after the underlying VM send syscall.
                 ErrorNumber::NotFound => {
-                    actor_error!(SysErrInvalidReceiver; "receiver not found")
+                    // How do we know the receiver is what triggered this?
+                    // An error number doesn't carry enough information at this level
+                    // above the raw syscall (and even then).
+                    actor_error!(not_found; "receiver not found")
```

Comment:
The send syscall is defined to return `NotFound` if, and only if, the receiver doesn't exist.

Reply:
https://github.com/filecoin-project/fvm-project/blob/724650bf92b239bb7d2079cde7d5bad121d7c79f/07-errors.md#L306